impedance/models/circuits/fitting.py

```diff
@@ -91,27 +127,24 @@ def circuit_fit(frequencies, impedances, circuit, initial_guess, constants,
     """
     Z = impedances
 
-    # extract the elements from the circuit
-    extracted_elements = extract_circuit_elements(circuit)
+    # pop kwargs passed in as kwargs
```

Comment:
I assume this is here so you can pass in a dictionary of kwargs in as `.fit(..., kwargs=kwargs)`, but I think it might be easier to just expand the dictionary when calling fit (e.g. `.fit(..., **kwargs)`). That way we don't need these lines and the **kwargs in circuit_fit will catch everything that's not a keyword parameter (e.g. not bounds, global_opt) and forward that on to the optimization

Reply:
Didn't know you could do that!